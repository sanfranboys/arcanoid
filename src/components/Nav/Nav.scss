.nav {
  &__list {
    list-style: none;
    display: flex;
    gap: 20px;
  }

  &__link {
    color: var(--color-gold);
    font-size: 16px;

    &:hover {
      color: var(--link-hover);
    }
    &.active {
      color: var(--color-light-purple);
    }
  }
}
