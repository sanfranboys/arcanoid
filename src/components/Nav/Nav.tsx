import React, { FC } from 'react'

import { NavLink } from 'react-router-dom'
import { Button } from 'elements'
import { toggleFullScreen } from 'utils'
import { root } from '@/constants'

import './Nav.scss'

const Nav: FC = () => (
  <nav className="nav">
    <ul className="nav__list">
      <li className="nav__item">
        <NavLink className="nav__link" to="/game">
          <Button>
            <span>Играть</span>
          </Button>
        </NavLink>
      </li>
      <li className="nav__item">
        <NavLink className="nav__link" to="/forum">
          <span>Форум</span>
        </NavLink>
      </li>
      <li className="nav__item">
        <NavLink className="nav__link" to="/leaderboard">
          <span>Лидеры</span>
        </NavLink>
      </li>
      <li className="nav__item">
        <NavLink className="nav__link" to="/profile">
          <span>Профиль</span>
        </NavLink>
      </li>
      <li className="nav__item nav__full-screen">
        <Button onClick={toggleFullScreen(root)}>Полноэкранный режим</Button>
      </li>
    </ul>
  </nav>
)

export default Nav
