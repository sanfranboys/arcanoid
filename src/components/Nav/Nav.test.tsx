import * as React from 'react'
import { BrowserRouter } from 'react-router-dom'
import * as renderer from 'react-test-renderer'
import Nav from './Nav'

describe('Nav testing', () => {
  it('Button renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
