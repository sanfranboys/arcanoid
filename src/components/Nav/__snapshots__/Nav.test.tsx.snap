// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav testing Button renders correctly 1`] = `
<nav
  className="nav"
>
  <ul
    className="nav__list"
  >
    <li
      className="nav__item"
    >
      <a
        aria-current={null}
        className="nav__link"
        href="/game"
        onClick={[Function]}
      >
        <button
          className="button-component"
          type="button"
        >
          <span>
            Играть
          </span>
        </button>
      </a>
    </li>
    <li
      className="nav__item"
    >
      <a
        aria-current={null}
        className="nav__link"
        href="/forum"
        onClick={[Function]}
      >
        <span>
          Форум
        </span>
      </a>
    </li>
    <li
      className="nav__item"
    >
      <a
        aria-current={null}
        className="nav__link"
        href="/leaderboard"
        onClick={[Function]}
      >
        <span>
          Лидеры
        </span>
      </a>
    </li>
    <li
      className="nav__item"
    >
      <a
        aria-current={null}
        className="nav__link"
        href="/profile"
        onClick={[Function]}
      >
        <span>
          Профиль
        </span>
      </a>
    </li>
    <li
      className="nav__item nav__full-screen"
    >
      <button
        className="button-component"
        onClick={[Function]}
        type="button"
      >
        Полноэкранный режим
      </button>
    </li>
  </ul>
</nav>
`;
